app/javascript/app/document-capture/components/file-input.jsx

```diff
@@ -89,8 +105,20 @@ export function toDataURL(file) {
   });
 }
 
+/**
+ * @type {import('react').ForwardRefExoticComponent<FileInputProps & ReactRefAttributes>}
+ */
 const FileInput = forwardRef((props, ref) => {
-  const { label, hint, bannerText, accept, value, errors, onClick, onChange } = props;
+  const {
+    label,
+    hint,
+    bannerText,
+    accept,
+    value,
+    errors = [],
+    onClick = () => {},
+    onChange = () => {},
```

Comment:
how come the `onChange` in this file and the other now have a default value? Because they're explicitly typed now?

Reply:
Now I see that `defaultProps` went away, so those must have been used and not just for decoration

Reply:
> Now I see that `defaultProps` went away, so those must have been used and not just for decoration

Yep, `defaultProps` is understood by React, where the values are used to assign a prop value if absent by the rendering parent. It was more useful with class-based components, since many instance functions within a class might want to reference the prop value, which could be error-prone if either the developer fails to assign defaults when referencing, or forgets to update it universally if the default ever changes. It's not as big a problem with function components, since these values are the sole source of truth for props referencing.

Generally speaking, I've seen a shift away from prop types from the core team, between the creation and adoption of [Flow](https://flow.org/) and the externalizing of the [`prop-types` package](https://www.npmjs.com/package/prop-types) (it was previously available as `React.PropTypes`). Depending how big a push there is toward advocating for function components, I could imagine a future where `propTypes` and `defaultProps` are removed altogether.

Reply:
To further clarify: React does validate `propTypes` in development environments. We do lose that with these changes. But we gain an equivalent checking through TypeScript instead. One which is arguably much better, since it occurs at build-time, not at run-time.